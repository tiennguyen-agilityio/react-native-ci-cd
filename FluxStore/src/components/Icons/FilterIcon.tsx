import {memo} from 'react';
import {TouchableOpacity} from 'react-native';
import Svg, {Ellipse, Path} from 'react-native-svg';

import {IconProps} from '@/interfaces';
import {useThemeStore} from '@/stores';

export const FilterIcon = memo(
  ({width = 27, height = 27, color, disabled = false, style, onPress}: IconProps) => {
    const {
      theme: {text},
    } = useThemeStore();

    return (
      <TouchableOpacity
        testID="filter-icon"
        activeOpacity={0.8}
        onPress={onPress}
        disabled={disabled}
        style={style}>
        <Svg width={width} height={height} viewBox="0 0 27 27" fill="none">
          <Path
            d="M12.375 9H22.5M4.5 18h11.25"
            stroke={color ?? text.tertiary}
            strokeWidth={2}
            strokeLinecap="round"
          />
          <Ellipse
            cx={7.875}
            cy={9}
            rx={3.375}
            ry={3.375}
            transform="rotate(90 7.875 9)"
            stroke={color ?? text.tertiary}
            strokeWidth={2}
            strokeLinecap="round"
          />
          <Ellipse
            cx={19.125}
            cy={18}
            rx={3.375}
            ry={3.375}
            transform="rotate(90 19.125 18)"
            stroke={color ?? text.tertiary}
            strokeWidth={2}
            strokeLinecap="round"
          />
        </Svg>
      </TouchableOpacity>
    );
  },
);
