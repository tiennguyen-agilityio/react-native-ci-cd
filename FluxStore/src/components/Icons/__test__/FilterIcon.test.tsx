import {render, fireEvent, screen} from '@testing-library/react-native';

import {FilterIcon} from '../FilterIcon';

describe('FilterIcon', () => {
  const onPress = jest.fn();

  it('renders correctly (matches snapshot)', () => {
    const {toJSON} = render(<FilterIcon />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('renders correctly (matches snapshot) with isActive', () => {
    const {toJSON} = render(<FilterIcon isActive />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onPress when touched and not disabled', () => {
    render(<FilterIcon onPress={onPress} />);

    fireEvent.press(screen.getByTestId('filter-icon'));

    expect(onPress).toHaveBeenCalledTimes(1);
  });

  it('does NOT call onPress when disabled', () => {
    render(<FilterIcon onPress={onPress} disabled />);

    fireEvent.press(screen.getByTestId('filter-icon'));

    expect(onPress).not.toHaveBeenCalledTimes(0);
  });
});
