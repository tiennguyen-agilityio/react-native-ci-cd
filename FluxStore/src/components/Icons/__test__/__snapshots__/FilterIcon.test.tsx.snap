// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterIcon renders correctly (matches snapshot) 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="filter-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={27}
    bbWidth={27}
    fill="none"
    focusable={false}
    height={27}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 27,
          "width": 27,
        },
      ]
    }
    vbHeight={27}
    vbWidth={27}
    width={27}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M12.375 9H22.5M4.5 18h11.25"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
          ]
        }
        stroke={
          {
            "payload": 4286021264,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeWidth={2}
      />
      <RNSVGEllipse
        cx={7.875}
        cy={9}
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        matrix={
          [
            6.123233995736766e-17,
            1,
            -1,
            6.123233995736766e-17,
            16.875,
            1.1249999999999991,
          ]
        }
        propList={
          [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
          ]
        }
        rx={3.375}
        ry={3.375}
        stroke={
          {
            "payload": 4286021264,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeWidth={2}
      />
      <RNSVGEllipse
        cx={19.125}
        cy={18}
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        matrix={
          [
            6.123233995736766e-17,
            1,
            -1,
            6.123233995736766e-17,
            37.125,
            -1.125,
          ]
        }
        propList={
          [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
          ]
        }
        rx={3.375}
        ry={3.375}
        stroke={
          {
            "payload": 4286021264,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeWidth={2}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;

exports[`FilterIcon renders correctly (matches snapshot) with isActive 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "opacity": 1,
    }
  }
  testID="filter-icon"
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight={27}
    bbWidth={27}
    fill="none"
    focusable={false}
    height={27}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "flex": 0,
          "height": 27,
          "width": 27,
        },
      ]
    }
    vbHeight={27}
    vbWidth={27}
    width={27}
  >
    <RNSVGGroup
      fill={null}
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M12.375 9H22.5M4.5 18h11.25"
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        propList={
          [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
          ]
        }
        stroke={
          {
            "payload": 4286021264,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeWidth={2}
      />
      <RNSVGEllipse
        cx={7.875}
        cy={9}
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        matrix={
          [
            6.123233995736766e-17,
            1,
            -1,
            6.123233995736766e-17,
            16.875,
            1.1249999999999991,
          ]
        }
        propList={
          [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
          ]
        }
        rx={3.375}
        ry={3.375}
        stroke={
          {
            "payload": 4286021264,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeWidth={2}
      />
      <RNSVGEllipse
        cx={19.125}
        cy={18}
        fill={
          {
            "payload": 4278190080,
            "type": 0,
          }
        }
        matrix={
          [
            6.123233995736766e-17,
            1,
            -1,
            6.123233995736766e-17,
            37.125,
            -1.125,
          ]
        }
        propList={
          [
            "stroke",
            "strokeWidth",
            "strokeLinecap",
          ]
        }
        rx={3.375}
        ry={3.375}
        stroke={
          {
            "payload": 4286021264,
            "type": 0,
          }
        }
        strokeLinecap={1}
        strokeWidth={2}
      />
    </RNSVGGroup>
  </RNSVGSvgView>
</View>
`;
